ziti/src/test/kotlin/io/netfoundry/ziti/net/nio/AsyncTLSChannelTest.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) 2018-2020 NetFoundry, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.netfoundry.ziti.net.nio
+
+import org.hamcrest.CoreMatchers.isA
+import org.hamcrest.CoreMatchers.startsWith
+import org.junit.After
+import org.junit.Assert.*
+import org.junit.Rule
+import org.junit.Test
+import org.junit.rules.ExpectedException
+import java.net.InetSocketAddress
+import java.nio.ByteBuffer
+import java.nio.channels.AsynchronousSocketChannel
+import java.nio.charset.StandardCharsets
+import java.util.concurrent.CompletableFuture
+import java.util.concurrent.ExecutionException
+import java.util.concurrent.TimeUnit
+import javax.net.ssl.SSLContext
+import javax.net.ssl.SSLException
+import kotlin.random.Random
+
+class AsyncTLSChannelTest {
+
+    lateinit var ch: AsynchronousSocketChannel
+
+    @Rule
+    @JvmField val thrown = ExpectedException.none()
+
+    @After
+    fun tearDown() {
+        ch.close()
+    }
+
+    @Test
+    fun connect() {
+        ch = AsyncTLSChannel.open()
+        ch.connect(InetSocketAddress("httpbin.org", 443)).get(1, TimeUnit.SECONDS)
```

Comment:
is the 1 the timeout? for sure some day 1 second won't be enough. for my money, i'd encourage more ... 5 or even 10 seconds. if that timeout hits - sure the build will be way slow but i think that's better than a false positive some day. also - 1 second would be a nice TIMEOUT constant seeing all the 1, TimeUnit.SECONDS on there :)